server/task.go

```diff
@@ -458,6 +458,20 @@ func (s *Server) composeTaskRelationship(ctx context.Context, raw *api.TaskRaw)
 		taskCheckRun.Updater = updater
 	}
 
+	blockedBy := []*api.Task{}
+	taskDAGList, err := s.store.FindTaskDAGList(ctx, &api.TaskDAGFind{FromTaskID: raw.ID})
+	if err != nil {
+		return nil, err
+	}
+	for _, taskDAG := range taskDAGList {
+		task, err := s.TaskService.FindTask(ctx, &api.TaskFind{ID: &taskDAG.ToTaskID})
+		if err != nil {
+			return nil, err
+		}
+		blockedBy = append(blockedBy, task.ToTask())
+	}
+	task.BlockedBy = blockedBy
```

Comment:
Will we get any inconsistency on tasks? These blockedBy tasks don't have the task dependency included.

Reply:
Yes, there is inconsistency on tasks, namely `blockedBy[i]` doesn't have its `blockedBy` field composed. However, I don't think we should compose every `blockedBy` field recursively because 1. the JSON message would get bloated and 2. frontend might not need that deep at all.